### PR TITLE
added type hinting in code examples

### PR DESCRIPTION
**Pull request by Yellow-Beans**
## Information

- [ ] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed).
- [x] This PR is **not** a code change (e.g. documentation, README, typehinting,
      examples, ...).

## Checklist

- [x] I have searched the open pull requests for duplicates.
- [ ] If code changes were made then they have been tested.
  - [ ] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why.
- [ ] I have updated the changelog to include these changes.

### CODE REVIEW

**Review by JustaSqu1d — CHANGES_REQUESTED** [review]
Could you also update the related explanations to explain the type-hinting then?

**Review by Yellow-Beans — COMMENTED** [review]
Added text, explaining the type hint

File: docs/getting-started/creating-your-first-bot.mdx [hunk 1]
```diff
@@ -200,6 +200,8 @@ decorator to define a slash command. We specify the `name` and `description` arg
 specified, the name of the slash command would be the function name and the command description would
 be empty.
 
+Optional: We typehint [': discord.ApplicationContext'](https://docs.pycord.dev/en/stable/api/application_commands.html#discord.ApplicationContext) to get access to autocompletion and docstrings.
```
Comment (JustaSqu1d) [inline docs/getting-started/creating-your-first-bot.mdx:203]:
```suggestion
Optional: We type-hint [`discord.ApplicationContext`](https://docs.pycord.dev/en/stable/api/application_commands.html#discord.ApplicationContext). In many modern IDEs, we can use this to access autocompletion and docstrings. You can read more about type-hinting [here](https://docs.python.org/3/library/typing.html).
```

File: docs/getting-started/creating-your-first-bot.mdx [hunk 2]
```diff
@@ -191,7 +191,7 @@ event that is automatically called when the bot is ready to use.
 
 ```py
 @bot.slash_command(name = "hello", description = "Say hello to the bot")
-async def say_hello(ctx):
+async def say_hello(ctx: discord.ApplicationContext):
```
Comment (JustaSqu1d) [inline docs/getting-started/creating-your-first-bot.mdx:194]:
```suggestion
async def hello(ctx: discord.ApplicationContext):
```

Not part of this PR, but this should be consistent with the actual code example lol.

**Review by JustaSqu1d — CHANGES_REQUESTED** [review]

**Review by JustaSqu1d — APPROVED** [review]

File: docs/getting-started/creating-your-first-bot.mdx (hunk 1 above)
Comment (Yellow-Beans) [reply docs/getting-started/creating-your-first-bot.mdx:203]: I'd use:
[...] we use this to access [...].
Without "can", because it's already marked as optional.

Otherwise I like your version better. appears more professional.

Comment (Dorukyum) [reply docs/getting-started/creating-your-first-bot.mdx:203]:
```suggestion
Optional: We type-hint the context parameter (`ctx`) as [`discord.ApplicationContext`](https://docs.pycord.dev/en/stable/api/application_commands.html#discord.ApplicationContext). In modern IDEs, type-hinting allows access to autocompletion and docstrings. You can read more about type-hinting [here](https://docs.python.org/3/library/typing.html).
```

**Review by Lulalaby — APPROVED** [review]
